server/consumer.go

```diff
@@ -5257,23 +5257,30 @@ func gatherSubjectFilters(filter string, filters []string) []string {
 	return filters
 }
 
-// Will check if we are running in the monitor already and if not set the appropriate flag.
-func (o *consumer) checkInMonitor() bool {
+// shouldStartMonitor will return true if we should start a monitor
```

Comment:
return true if we did start the monitor routine.

Reply:
At this point we haven't actually started it yet, the caller to `shouldStartMonitor` is where the `startGoRoutine` call is.

Reply:
Apologies you are correct.